src/v/net/probes.cc

```diff
@@ -282,4 +299,187 @@ std::ostream& operator<<(std::ostream& o, const client_probe& p) {
       << ", requests_blocked_memory: " << p._requests_blocked_memory << " }";
     return o;
 }
+
+ss::future<ss::shared_ptr<ss::tls::server_credentials>>
+build_reloadable_server_credentials_with_probe(
+  config::tls_config config,
+  std::string_view service,
+  std::string_view listener_name,
+  ss::tls::reload_callback cb) {
+    auto builder = co_await config.get_credentials_builder();
+    if (!builder) {
+        co_return nullptr;
+    }
+    co_return co_await build_reloadable_credentials_with_probe<
+      ss::tls::server_credentials>(
+      std::move(*builder), service, listener_name, cb);
```

Comment:
nitpick: `cb` can be moved. Maybe the names if they become strings rather than views.